JSON Output

driftingly opened the pull request:
This PR starts the process of generating the output files. 

> [!NOTE]  
> This PR does not cover: 
> - limiting the output to only published themes, prompts, artworks.
> - Only using the top 8 on view artworks
> - Ordering prompts
> 
> Please call out anything else I may be missing. 

The URLs will stay the same at:
 - `/json/data.json`
 - `/json/data-es.json`
 - `/json/data-zh-hans.json`

The output is generated in real-time, taking between 10 and 20 seconds, depending on caching. In a follow-up PR the output will be written to disk and refreshed every 5 minutes.

The API will act as the source for any fields that do not change or fields required for CMS functionality.

The following fields are stored locally:
- `title` - Translated
- `artist_display` - Translated
- `is_on_view` - Displayed on the index page
- `image_id` - Used to display images on the index page

Additional fields are fetched through the API as needed.

**API Optimizations and Caching**

I added a method to query and cache all artwork to optimize the production of the output JSON files. This is done using chunks of 100, caching the objects using the single endpoint as a key. This caching strategy simplifies the JSON resources by preventing the need to pass API data down to individual resources. 

**Activity Templates**

Activity templates have been added using the [Sushi](https://usesushi.dev/) package to be treated as models within the application.

nikhiltri commented on routes/web.php:
```diff
@@ -1,7 +1,10 @@
     return redirect('/admin');
 });
+
+Route::get('/sites/default/files/json/{id}.json', JsonController::class);
```
I don't know that this path needs to match the Drupal conventions. We'll be updating the FE URL in its entirety, so make this a path that makes logical sense in the context of this app.

driftingly replied:
Updated to `/json/{id}.json`

nikhiltri reviewed (approved):
Wow, this is a lot! It looks good at first glance. If we want to get these updates into the hands of the users, I'm happy to merge this in. Left one small note, and a few other minor ones:

* On the artworks edit page, the main reference number readonly text is _much_ lighter than the Datahub ID text. It’s almost imperceptible. They’re defined the same way in the controller, so I’m not sure why one would be rendered differently.
* On the artworks edit page, instead of “Datahub ID” I think we can call that field “Object ID”. That's the language CMS users have been using when referring to that data.

zachgarwood reviewed (approved):
😎

driftingly commented:
> Wow, this is a lot! It looks good at first glance. If we want to get these updates into the hands of the users, I'm happy to merge this in. Left one small note, and a few other minor ones:
> 
> * On the artworks edit page, the main reference number readonly text is _much_ lighter than the Datahub ID text. It’s almost imperceptible. They’re defined the same way in the controller, so I’m not sure why one would be rendered differently.
> * On the artworks edit page, instead of “Datahub ID” I think we can call that field “Object ID”. That's the language CMS users have been using when referring to that data.

Thanks @nikhiltri 🙌 

I updated "Datahub ID" to "Object ID".

The issue with the text being lighter was only an issue on fields not stored in the DB. Since we are not saving these fields, I updated that section to be a simple blade view with no form fields.

<img width="932" alt="Screenshot 2024-04-24 at 12 29 14 PM" src="https://github.com/art-institute-of-chicago/journeymaker-cms/assets/194221/e0af4e27-0247-4ee5-ad9a-f5465d4afb06">
